pkg/boot/universalpayload/handoffblock.go

```diff
@@ -0,0 +1,242 @@
+	EFIResourceAttributeMultipleBitECC        EFIResourceAttributeType = 0x00000010
+	EFIResourceAttributeECCReserved1          EFIResourceAttributeType = 0x00000020
+	EFIResourceAttributeECCReserved2          EFIResourceAttributeType = 0x00000040
+	EFIResourceAttributeUncacheable           EFIResourceAttributeType = 0x00000400
+	EFIResourceAttributeWriteCombineable      EFIResourceAttributeType = 0x00000800
+	EFIResourceAttributeWriteThroughCacheable EFIResourceAttributeType = 0x00001000
+	EFIResourceAttributeWriteBackCacheable    EFIResourceAttributeType = 0x00002000
+	EFIResourceAttribute16BitIO               EFIResourceAttributeType = 0x00004000
+	EFIResourceAttribute32BitIO               EFIResourceAttributeType = 0x00008000
+	EFIResourceAttribute64BitIO               EFIResourceAttributeType = 0x00010000
+	EFIResourceAttributeUncachedExported      EFIResourceAttributeType = 0x00020000
+	EFIResourceAttributeReadProtectable       EFIResourceAttributeType = 0x00100000
+	EFIResourceAttributeWriteProtectable      EFIResourceAttributeType = 0x00200000
+	EFIResourceAttributeExecutionProtectable  EFIResourceAttributeType = 0x00400000
+	EFIResourceAttributePersistable           EFIResourceAttributeType = 0x01000000
+	EFIResourceAttributeReadOnlyProtected     EFIResourceAttributeType = 0x00040000
+	EFIResourceAttributeReadOnlyProtectable   EFIResourceAttributeType = 0x00080000
+	EFIResourceAttributeEncrypted             EFIResourceAttributeType = 0x04000000
+	EFIResourceAttributeSpecialPurpose        EFIResourceAttributeType = 0x08000000
+	EFIResourceAttributeMoreReliable          EFIResourceAttributeType = 0x02000000
+)
+
+type EFIHOBHandOffBootModeType uint32
+type EFIHOBHandoffInfoVersionType uint32
+
+// EFIHOBHandoffInfoTable values
+const (
+	EFIHOBHandoffInfoVersion          EFIHOBHandoffInfoVersionType = 0x09
+	EFIHOBHandoffInfoBootMode         EFIHOBHandOffBootModeType    = 0x0
+	EFIHOBHandoffInfoEFIMemoryTop     EFIPhysicalAddress           = 0x4E00000
+	EFIHOBHandoffInfoEFIMemoryBottom  EFIPhysicalAddress           = 0x800000
+	EFIHOBHandoffInfoFreeEFIMemoryTop EFIPhysicalAddress           = 0x4E00000
+	EFIHOBHandoffInfoFreeMemoryBottom EFIPhysicalAddress           = 0xE00000
+)
+
+// EFIHOBHandoffInfoTable appears in the first in HOB list, which contains some general information
+type EFIHOBHandoffInfoTable struct {
+	Header           EFIHOBGenericHeader
+	Version          EFIHOBHandoffInfoVersionType
+	BootMode         EFIHOBHandOffBootModeType
+	MemoryTop        EFIPhysicalAddress
+	MemoryBottom     EFIPhysicalAddress
+	FreeMemoryTop    EFIPhysicalAddress
+	FreeMemoryBottom EFIPhysicalAddress
+	EndOfHOBList     EFIPhysicalAddress
+}
+
+// EFIHOBResourceDescriptor describes all resource properties found on the processor host bus
+type EFIHOBResourceDescriptor struct {
+	Header            EFIHOBGenericHeader
+	Owner             [16]byte // EFIGUID (assuming a byte array for simplicity)
```

Comment:
Below in `EFIHOBGUIDType`, you're using `guid.UUID` for this, but here `[16]byte`?